### PR TITLE
[STORM-1529] Change default temp dir for workers to worker launch directory.

### PR DESCRIPTION
**Pull request by kishorvpatil**
Allowing workers to create temp files unde  /tmp/ directory creates different challenges for monitoring disk usage and cleanup. So making "tmp" directory under workers directory, so it gets cleaned up in case of any deletion.

### CODE REVIEW

**Comment by revans2** [comment]
+1 setting the tmp directory to something tied to the lifetime of the worker should avoid leaks.

**Comment by revans2** [comment]
Either storm-hdfs is having some issues of it's own or it does not like this change.  I think it is probably that it is unstable.

File: storm-core/test/clj/org/apache/storm/supervisor_test.clj [hunk 1]
```diff
@@ -299,6 +299,7 @@
                                 "-Dstorm.conf.file="
                                 "-Dstorm.options="
                                 (str "-Dstorm.log.dir=" file-path-separator "logs")
+                                (str "-Djava.io.tmpdir=/tmp/workers" file-path-separator mock-worker-id file-path-separator "tmp")
```
Comment (d2r) [inline storm-core/test/clj/org/apache/storm/supervisor_test.clj]: Using `/` will break tests on windows.

Comment (d2r) [inline storm-core/test/clj/org/apache/storm/supervisor_test.clj]: Actually, this is not what is under test here, so I stand corrected. The code that would use this path is stubbed/mocked I think.

**Comment by kishorvpatil** [comment]
@revans2 It does not look like its related. The hdfs tests have been for some time now. I noticed for #1075, the !storm-core builds failed - https://travis-ci.org/apache/storm/jobs/107157830

**Comment by knusbaum** [comment]
@kishorvpatil Needs upmerge.

**Comment by kishorvpatil** [comment]
@knusbaum, I addressed the conflicts.

**Comment by knusbaum** [comment]
+1

**Comment by ptgoetz** [comment]
needs another upmerge.

**Comment by kishorvpatil** [comment]
@ptgoetz upmerged.

**Comment by kishorvpatil** [comment]
@revans2 @ptgoetz I have upmerged this again. I feel this is important fix to avoid letting workers write to /tmp and cause disk full issues. So we should put this in 1.x-branch as well. 
If no objections, I will pull it into 1.x-branch as well.

**Comment by ptgoetz** [comment]
+1 for merging to both branches.

**Comment by kishorvpatil** [comment]
Travis CI build failure is unrelated

**Comment by kishorvpatil** [comment]
@revans2 @ptgoetz 
I created a separate Pull request for 1.x-branch as I can't merge ConfigUtils.  #1187. Please review.

**Pull request merged**